main.py

```diff
@@ -44,14 +45,14 @@ def handle_pubsub_message():
     try:
         pubsub_data_string = pubsub.parse_data_from_message(pubsub_received_message)
     except pubsub.DataParseError as e:
-        print(e)
+        app.logger.error(e)
         return (str(e), 400)
 
     # load the notification from the data
     try:
         monitoring_notification_dict = json.loads(pubsub_data_string)
     except json.JSONDecodeError as e:
-        print(e)
+        app.logger.error(e)
```

Comment:
Can you research how we might log in other modules in this same application?

For instance, what if we want to log in philips_hue.py?

Does the Flask logger "point" to an underlying Python logger. So then if we call logging.getLogger(name) in another module, we will get the same logger by default? Or does something else need to be done manage logging in other modules?

Either way, I think it's a good idea to use the flask logger, since apparently it automatically logs things like requests and other server events.

Reply:
After researching and experimenting on my machine, I found that you were right in that you can just call logging.getLogger(name) (where name is the name of the flask application) in another module to access the flask logger. However, as of now we are using __name__ as the name of the flask app whose value isn't consistent throughout other modules making it hard to access the logger.

I thought that we could just have the app name be in the config file in order to be accessible, but this wouldn't work since the app has to be already initialized to access the config values. I see a potential option as just creating an application name variable outside of the config classes in the config.py file that can be easily imported by all modules. Do you think this would be a good approach?

On another note, I found that python allows the creation of child loggers by using a structure like logging.getLogger(name.child) where the child logger logs to the same destinations as the parent logger. I think this could be useful in our case by creating a child logger in other modules (called flask_name.module_name) to make it clearer what module the logs are coming from.

Reply:
Discussed offline:

We can just call `logger.get_logger()` in other modules to get the top-level Python logger. The Flask logger is a separate logger that inherits defaults from the top-level logger.

Reply:
Done